### PR TITLE
pre-alpha flux build and requisite dependencies

### PR DESCRIPTION
**Pull request by trws**
NOTE: the ncurses package now installs *BOTH* versions, wide and non-wide
character for compatibility with packages that require either one.  It may be
desirable for this to be an option in the future.

### CODE REVIEW

File: var/spack/packages/flux/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    homepage = "https://github.com/flux-framework/flux-core"
+    url      = "https://github.com/flux-framework/flux-core"
+
+    version('master', branch='master', git='https://github.com/flux-framework/flux-core')
+
+    # Also needs autotools, but should use the system version if available
+    depends_on("zeromq@4.0.4:")
+    depends_on("czmq@2.2:")
+    depends_on("lua@5.1:5.1.99")
```
Comment (tgamblin) [inline var/spack/packages/flux/package.py]: @trws: did @5.1 not work?  "@5.1" should match 5.1.99, but there may be a bug in the way this is handled in the concretizer.

**Comment by tgamblin** [comment]
@trws: thanks!

File: var/spack/packages/flux/package.py (hunk 1 above)
Comment (trws) [inline var/spack/packages/flux/package.py]: It should work, but flux can't tolerate lua 5.2+, so I wanted to make sure it was only within the 5.1 range.  Is there a better way to mask off the high end?

Comment (tgamblin) [inline var/spack/packages/flux/package.py]: Just "@5.1" should match any 5.1.x.  It should look at that and try to match it against known versions, then pick one.  So if you depend on lua@5.1, the intended semantics is to take the highest available 5.1.  In the future that should change to "preferred 5.1", and packages and config files should be able to say which one they prefer.

**Pull request merged**